plugins/civo/api_key.go

```diff
@@ -0,0 +1,101 @@
+package civo
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://www.civo.com/docs/account/api-keys"),
+		ManagementURL: sdk.URL("https://dashboard.civo.com/security"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "API Key used to authenticate to Civo.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 50,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.APIKeyID,
+				MarkdownDescription: "API Name to identify the API Key.",
+			},
+			{
+				Name:                fieldname.DefaultRegion,
+				MarkdownDescription: "The default region to use for this API Key.",
+				Optional:            true,
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TryCivoConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"CIVO_TOKEN":        fieldname.APIKey,
+	"CIVO_API_KEY_NAME": fieldname.APIKeyID,
+	"CIVO_API_KEY":      fieldname.APIKey,
```

Comment:
Let's separate these in 2 maps: one for `CIVO_TOKEN` and one for the other two variables.

Reply:
suree